scripts/build/dependencies-npm.sh

```diff
@@ -96,6 +96,18 @@ if [ "$ARGV_PRODUCTION" == "true" ]; then
 fi
 
 function run_install() {
+
+  # Since we use an `npm-shrinkwrap.json` file, if you pull changes
+  # that update a dependency and try to `npm install` directly, npm
+  # will complain that your `node_modules` tree is not equal to what
+  # is defined by the `npm-shrinkwrap.json` file, and will thus
+  # refuse to do anything but install from scratch.
+  npm prune
+
+  # When changing between target architectures, rebuild all dependencies,
+  # since compiled add-ons will not work otherwise.
+  npm rebuild
```

Comment:
Could we optimise this further to only rebuild if the new target-arch is different to the old target-arch?